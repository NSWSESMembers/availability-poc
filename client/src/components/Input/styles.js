import { StyleSheet } from 'react-native';
import Colors from '../../themes/Colors';

const styles = StyleSheet.create({
  default: {
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    color: Colors.bgMain,
    fontWeight: '700',
    fontSize: 20,
    flex: 1,
  },
  defaultHolder: {
    borderBottomWidth: 0.5,
    borderColor: Colors.bdInput,
    marginBottom: 25,
    width: '100%',
  },
  defaultText: {
    color: Colors.bgMain,
    fontWeight: '700',
  },
  inputHolder: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export default styles;
