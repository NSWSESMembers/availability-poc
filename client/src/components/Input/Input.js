import React from 'react';
import { View, TextInput } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

const Input = (props) => {
  const { value, placeholder, type, onChangeText } = props;

  const propsForType = {
    username: {
      autoCapitalize: 'none',
      autoCorrect: false,
      returnKeyType: 'next',
    },
    email: {
      autoCapitalize: 'none',
      autoCorrect: false,
      keyboardType: 'email-address',
      returnKeyType: 'next',
    },
    password: {
      secureTextEntry: true,
    },
  };

  let typeProps = propsForType[type];

  if (typeof typeProps === 'undefined') {
    typeProps = {};
  }

  return (
    <View style={styles.defaultHolder}>
      <View style={styles.inputHolder}>
        <TextInput
          onChangeText={onChangeText}
          {...typeProps}
          value={value}
          placeholder={placeholder}
          underlineColorAndroid="rgba(0,0,0,0)"
          style={styles.default}
          placeholderTextColor="#C2C2C2"
        />
      </View>
    </View>
  );
};

Input.propTypes = {
  type: PropTypes.string,
  title: PropTypes.string,
  value: PropTypes.string,
  valid: PropTypes.bool,
  placeholder: PropTypes.string,
  onChangeText: PropTypes.func,
};

export default Input;
