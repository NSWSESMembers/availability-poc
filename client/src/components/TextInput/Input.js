import React from 'react';
import { View, Text, TextInput } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const Input = (props) => {
  const { title, value, placeholder, type, onChangeText } = props;

  let typeProps = {};
  switch (type) {
    case 'email':
      typeProps = {
        autoCapitalize: 'none',
        autoCorrect: false,
        keyboardType: 'email-address',
        returnKeyType: 'next',
      };
      break;
    case 'password':
      typeProps = {
        secureTextEntry: true,
      };
      break;
    default:
      break;
  }
  return (
    <View>
      {title && <Text>{title}</Text>}
      <View style={styles.inputHolder}>
        <TextInput
          onChangeText={onChangeText}
          {...typeProps}
          value={value}
          placeholder={placeholder}
          underlineColorAndroid="rgba(0,0,0,0)"
          style={styles.default}
        />
      </View>
    </View>
  );
};

Input.propTypes = {
  type: PropTypes.string,
  title: PropTypes.string,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  onChangeText: PropTypes.func,
};

export default Input;
