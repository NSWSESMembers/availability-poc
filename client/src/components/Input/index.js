import Input from './Input';
import styles from './styles';

export { Input, styles };
