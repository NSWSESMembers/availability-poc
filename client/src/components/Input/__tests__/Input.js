import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../Input';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <Input
      type="username"
      title="title"
      value="value"
      placeholder="placeholder"
      onChangeText={noop}
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <Input
      type="email"
      title="title"
      value="value"
      placeholder="placeholder"
      onChangeText={noop}
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();

  const tree3 = renderer.create(
    <Input
      type="password"
      title="title"
      value="value"
      placeholder="placeholder"
      onChangeText={noop}
    />,
  ).toJSON();
  expect(tree3).toMatchSnapshot();

  const tree4 = renderer.create(
    <Input
      type="unknown"
      title="title"
      value="value"
      placeholder="placeholder"
      onChangeText={noop}
    />,
  ).toJSON();
  expect(tree4).toMatchSnapshot();
});
