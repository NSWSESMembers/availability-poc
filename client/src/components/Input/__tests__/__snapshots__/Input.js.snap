// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0.5,
      "borderColor": "#cbcbcb",
      "marginBottom": 25,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#C2C2C2"
      returnKeyType="next"
      style={
        Object {
          "alignItems": "center",
          "color": "#1e3c78",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 20,
          "fontWeight": "700",
          "height": 50,
          "justifyContent": "center",
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="value"
    />
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0.5,
      "borderColor": "#cbcbcb",
      "marginBottom": 25,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#C2C2C2"
      returnKeyType="next"
      style={
        Object {
          "alignItems": "center",
          "color": "#1e3c78",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 20,
          "fontWeight": "700",
          "height": 50,
          "justifyContent": "center",
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="value"
    />
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0.5,
      "borderColor": "#cbcbcb",
      "marginBottom": 25,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#C2C2C2"
      secureTextEntry={true}
      style={
        Object {
          "alignItems": "center",
          "color": "#1e3c78",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 20,
          "fontWeight": "700",
          "height": 50,
          "justifyContent": "center",
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="value"
    />
  </View>
</View>
`;

exports[`renders correctly 4`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0.5,
      "borderColor": "#cbcbcb",
      "marginBottom": 25,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#C2C2C2"
      style={
        Object {
          "alignItems": "center",
          "color": "#1e3c78",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 20,
          "fontWeight": "700",
          "height": 50,
          "justifyContent": "center",
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="value"
    />
  </View>
</View>
`;
